### PR TITLE
Query Metrics

### PR DESCRIPTION
**Pull request by ammuv**
Added histograms for query scans and returns.

### CODE REVIEW

File: src/k2/module/k23si/Module.cpp [hunk 1]
```diff
@@ -585,7 +585,7 @@ K23SIPartitionModule::handleQuery(dto::K23SIQueryRequest&& request, dto::K23SIQu
 
     IndexerIterator key_it = _initializeScan(request.key, request.reverseDirection, request.exclusiveKey);
     for (; !_isScanDone(key_it, request, response.results.size());
-                        _scanAdvance(key_it, request.reverseDirection, request.key.schemaName)) {
+                        _scanAdvance(key_it, request.reverseDirection, request.key.schemaName), numScans++) {
```
Comment (jfunston) [inline src/k2/module/k23si/Module.cpp:588]: I think it is better to move the increment into the for loop body, it is easy to miss here

File: src/k2/module/k23si/Module.cpp [hunk 2]
```diff
@@ -646,12 +646,12 @@ K23SIPartitionModule::handleQuery(dto::K23SIQueryRequest&& request, dto::K23SIQu
         request.key = key_it->first; // if we retry, do so with the key we're currently iterating on
         return _doPush(request.key, versions.WI->data.timestamp, request.mtr, deadline)
```
Comment (jfunston) [inline src/k2/module/k23si/Module.cpp:647]: I think maybe we shouldn't track numScans across pushes. The idea is to get a feel for how much the reactor is monopolized by the query, and the push will yield it. We can see what Ivan thinks.

**Review by jfunston — CHANGES_REQUESTED** [review]

File: src/k2/module/k23si/Module.cpp (hunk 1 above)
Comment (ammuv) [reply src/k2/module/k23si/Module.cpp:592]: done.

File: src/k2/module/k23si/Module.cpp (hunk 2 above)
Comment (ammuv) [reply src/k2/module/k23si/Module.cpp:658]: done.

**Review by jfunston — DISMISSED** [review]

**Review by iafuture — APPROVED** [review]

**Pull request merged**